src/runtimes/docker/contracts/ContainerClient.ts

```diff
@@ -334,13 +334,27 @@ export type BuildImageCommandOptions = CommonCommandOptions & {
     /**
      * Target platform for the image build
      */
-    platform?: string;
+    platform?: ContainerPlatform;
     /**
      * Additional custom options to pass
      */
     customOptions?: string;
 };
 
+/**
+ * Target platform for the image build
+ */
+export type ContainerPlatform = undefined | {
```

Comment:
I would define this type before BuildImageCommandOptions where it's first used (move it up in the file).